test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -763,10 +769,8 @@ void UcxConnection::stream_recv_callback(void *request, ucs_status_t status,
         assert(status == UCS_ERR_CANCELED);
     }
 
-    assert(conn->_establish_cb == r->callback);
-    common_request_callback(request, status);
-    delete conn->_establish_cb;
-    conn->_establish_cb = NULL;
```

Comment:
maybe keep `conn->_establish_cb = NULL;`?

Reply:
I moved it to `established` and `handle_connection_error` where the callback is invoked